tasks/build/package_scripts/post_remove.sh

```diff
@@ -9,10 +9,34 @@ user_remove() {
   userdel "$1"
 }
 
+REMOVE_USER=false
+
 case $1 in
-    purge|0)
-        if user_check "<%= user %>"  ; then
-          user_remove "<%= user %>"
-        fi
-        ;;
+
+  # Debian
+  purge)
+    REMOVE_USER=true
+  ;;
+
+  remove|failed-upgrade|abort-install|abort-upgrade|disappear|upgrade|disappear)
+  ;;
+
+  # Red Hat
+  0)
+    REMOVE_USER=true
+  ;;
+
+  1)
+  ;;
+
+  *)
+      echo "post remove script called with unknown argument \`$1'" >&2
+      exit 1
+  ;;
 esac
+
+if [ "$REMOVE_USER" = "true" ]; then
```

Comment:
You can simplify this to `if $REMOVE_USER; then` (note the lack of `[` and `]`). Taken from https://github.com/aperezdc/perezdecastro.org/blob/master/stash/using-boolean-variables-in-bash.markdown.